src/plugins/kibana_utils/common/persistable_state/migrate_to_latest.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { compare } from 'semver';
+import { SerializableState, VersionedState, MigrateFunctionsObject } from './types';
+
+export function migrateToLatest<S extends SerializableState>(
+  migrations: MigrateFunctionsObject,
+  [state, oldVersion]: VersionedState
+): VersionedState<S> {
+  const versions = Object.keys(migrations || {})
+    .sort(compare)
+    .filter((v) => compare(v, oldVersion) > 0);
```

Comment:
Flip for performance:

```suggestion
    .filter((v) => compare(v, oldVersion) > 0)
    .sort(compare);
```